server/internal/embedder/embedder.go

```diff
@@ -136,3 +137,27 @@ func splitFile(ctx context.Context, fileName, fileType string, chunkSizeTokens,
 		return nil, fmt.Errorf("unexpected file type: fileType=%q", fileType)
 	}
 }
+
+// DeleteFile deletes a file from the embedder.
+func (e *E) DeleteFile(ctx context.Context, collectionName, fileID string) error {
+	return e.vstoreClient.DeleteDocuments(ctx, collectionName, fileID)
+}
+
+// Search searches for the matched documents in the embedder for the given query.
+func (e *E) Search(ctx context.Context, collectionName, modelName, query string, numDocs int) ([]string, error) {
```

Comment:
No need to change now, but it might make sense to rename the struct and package as we have more functionality than embedding. This struct is currently an integration point for Milvus and Ollama.